pkg/types/conversion/installconfig.go

```diff
@@ -176,6 +176,19 @@ func convertOpenStack(config *types.InstallConfig) error {
 		return err
 	}
 
+	// machinesSubnet has been deprecated in favor of ControlPlanePort
+	controlPlanePort := config.Platform.OpenStack.ControlPlanePort
+	deprecatedMachinesSubnet := config.Platform.OpenStack.DeprecatedMachinesSubnet
+	if deprecatedMachinesSubnet != "" && controlPlanePort == nil {
+		fixedIPs := []openstack.FixedIP{{Subnet: openstack.SubnetFilter{ID: deprecatedMachinesSubnet}}}
+		config.Platform.OpenStack.ControlPlanePort = &openstack.PortTarget{FixedIPs: fixedIPs}
+	} else if deprecatedMachinesSubnet != "" &&
+		controlPlanePort != nil {
+		if !(len(controlPlanePort.FixedIPs) == 1 && controlPlanePort.FixedIPs[0].Subnet.ID == deprecatedMachinesSubnet) {
```

Comment:
Is this check really required? Should we be reporting an error when `deprecatedMachinesSubnet != "" &&
controlPlanePort != nil`? If so, can this test be a part of installconfig validation? We can report an error much earlier in the case of this mis-configuration.

Reply:
I believe it's required because if the user provided `machinesSubnet` then ran a `openshift-install --dir "$dir" create install-config` the install-config would get the `controlPlanePort`  added.

Similar check also happens for [the VIPs conversion](https://github.com/openshift/installer/blob/master/pkg/types/conversion/installconfig.go#L227-L231).